docs/architecture.md

```diff
@@ -0,0 +1,30 @@
+# Architecture
+Partial class diagram illustrating the important classes and functions for processing incoming blocks and relaying the contained Warp messages.
+```mermaid
+classDiagram
+    Listener o-- "per RelayerID" ApplicationRelayer
+    Listener *-- "1" Subscriber
```

Comment:
What does the "1" represent here in the composition link?

Reply:
For clarity, the same notation applies to compositions as well as aggregations. In the above example, `Listeners` aggregate `ApplicationRelayers`